rclcpp/include/rclcpp/node.hpp

```diff
@@ -1448,6 +1449,51 @@ class Node : public std::enable_shared_from_this<Node>
   rclcpp::node_interfaces::NodeTimeSourceInterface::SharedPtr
   get_node_time_source_interface();
 
+  /// Return a NodeInterfaces bound with the Node's internal node interfaces.
+  /**
+   * Specify which interfaces you want to bind using the template parameters by specifying
+   * interface support classes to use. Any unmentioned interfaces will be unavailable to bind.
+   *
+   * This method will return a NodeInterfaces with no bound interfaces. You must set them using
+   * ```NodeInterfaces->set_<interface_name>_interface(InterfaceT::SharedPtr interface)```
+   *
+   * You may use any of the available support classes in
+   * node_interfaces/node_interfaces_helpers.hpp:
+   *   - Base:       Supports NodeBaseInterface
+   *   - Clock:      Supports NodeClockInterface
+   *   - Graph:      Supports NodeGraphInterface
+   *   - Logging:    Supports NodeLoggingInterface
+   *   - Parameters: Supports NodeParametersInterface
+   *   - Services:   Supports NodeServicesInterface
+   *   - TimeSource: Supports NodeTimeSourceInterface
+   *   - Timers:     Supports NodeTimersInterface
+   *   - Topics:     Supports NodeTopicsInterface
+   *   - Waitables:  Supports NodeWaitablesInterface
+   *
+   * Or you can define your own interface support classes!
+   *
+   * Each of the support classes should define:
+   *   - Default constructor
+   *   - Templated constructor taking NodeT
+   *   - get_node_<interface_name>_interface()
+   *   - set_node_<interface_name>_interface()
+   *
+   * Usage example:
+   *   - ```NodeInterfaces<rclcpp::node_interfaces::Base>(node)```
+   *     will bind just the NodeBaseInterface.
+   *   - ```NodeInterfaces<
+   *          rclcpp::node_interfaces::Base, rclcpp::node_interfaces::Clock>(node)```
+   *     will bind both the NodeBaseInterface and NodeClockInterface.
```

Comment:
I think that these examples should be in `NodeInterfaces` documentation and not here

Reply:
[ros2/rclcpp@`2d65b59` (#2041)](https://github.com/ros2/rclcpp/pull/2041/commits/2d65b59c855d73b66ecbdf5638b59d408fc7f7b9)